lib/client/local_proxy_middleware.go

```diff
@@ -76,7 +75,7 @@ func NewAppCertChecker(tc *TeleportClient, appRoute proto.RouteToApp, clock cloc
 
 // OnNewConnection is a callback triggered when a new downstream connection is
 // accepted by the local proxy.
-func (c *CertChecker) OnNewConnection(ctx context.Context, lp *alpnproxy.LocalProxy, conn net.Conn) error {
```

Comment:
If it's possible to extract local VNet app address from `net.Conn`, it'd be beneficial to include it here. When I was adding telemetry, I ran into an issue where I cannot tell the user which app they were trying to access, because `types.Application` alone doesn't have this info obviously.

OTOH, wouldn't `net.Conn` here in the context of VNet be the connection between a local proxy and the proxy service? 🤔 So it still wouldn't tell us whether someone tried to open `app.cluster.com` or `app.custom-dns-zone.com`.

Anyway, I can handle adding this down the line when I'll be working on showing recent connections in Connect.

Reply:
What do you mean "I cannot tell the user which app they were trying to access, because types.Application alone doesn't have this info obviously"? Seems like you have all the info you'd need about which app is being accessed here https://github.com/gravitational/teleport/pull/41587

but no, i don't think you could get any extra info from the `net.Conn` here other than the client-side and vnet IP address, but I don't see why we'd need those. We wouldn't be able to get the DNS name the user is trying to connect to from the conn

Reply:
Sorry, I should've said "which DNS name" and not "which app they were trying to access".

Displaying the DNS name will be useful once I get around to showing connections in the UI.

Reply:
the only valid DNS name is the app's public_addr though

Reply:
I was thinking about edge cases such as an app with a custom `public_addr`, say `example.com`, being accessed over `app.cluster.com` anyways – this should still work, right? In that, `types.Application.GetPublicAddr` would return `example.com` instead of `app.cluster.com`.